npm_and_yarn/lib/dependabot/npm_and_yarn/package_manager.rb

```diff
@@ -1,19 +1,59 @@
 # typed: true
 # frozen_string_literal: true
 
+require "dependabot/shared_helpers"
+
 module Dependabot
   module NpmAndYarn
     class PackageManager
-      def initialize(package_json)
+      def initialize(package_json, lockfiles:)
         @package_json = package_json
+        @lockfiles = lockfiles
+        @package_manager = package_json.fetch("packageManager", nil)
+      end
+
+      def setup(name)
+        return unless @package_manager.nil? || @package_manager.start_with?("#{name}@")
+
+        version = requested_version(name)
+
+        if version
+          install(name, version)
+        else
+          version = guessed_version(name)
+
+          if version && name == "pnpm"
+            if Version.new(version.to_s) < Version.new("7")
```

Comment:
Is there any way the requested_version could be < 7? Should we account for this there as well?

Reply:
Yes, and I think this set of changes is unintentionally restoring support for that. We should definitely be guarding and raising in both cases, I'll fix this.